Add localcan extension

ajzbc opened the pull request:
## Description

Lists domains in [LocalCan](https://localcan.com) and provide shortcuts to open, copy, etc

## Screencast

LocalCan:
![image](https://github.com/raycast/extensions/assets/13984226/a8bc7436-08d4-4c82-8d02-cc28d521c3d4)

Raycast Extension:
![local-can-example](https://github.com/raycast/extensions/assets/13984226/3345002d-a5b3-4731-9174-af410458ac53)


## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- If you add `Domain only`  it renders a lot of `null` in the item

![Raycast19012024-pbNNbVjS png@2x](https://github.com/raycast/extensions/assets/38033107/72676d71-1af5-499a-afee-a7571cf225b4)

- Could you maybe update the toast to include a installation URL, check how extensions like [Eagle](https://www.raycast.com/Yukai/eagle) does it

- Maybe popToRoot() on error or add something in the Detail pane instead of showing nothing

![Raycast 2024-01-19 at 14 20 25](https://github.com/raycast/extensions/assets/38033107/547300bd-6e71-41da-9b6f-0ed9393c3d47)

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

ajzbc commented:
@pernielsentikaer thanks for the review!

Added support for "Domain Only" (Sorry totally missed that before 😅)

![image](https://github.com/raycast/extensions/assets/13984226/336197b1-cb60-46bc-82ba-d0d5d11bf155)

Added a `<Detail>` to show the error

![image](https://github.com/raycast/extensions/assets/13984226/df815ae3-6cb3-465b-9059-4341892f4b58)

pernielsentikaer reviewed (approved):
Hi 👋 

Looks good to me, approved 🔥
